src/main/java/com/gmail/nossr50/datatypes/skills/alchemy/AlchemyPotion.java

```diff
@@ -49,21 +52,26 @@ public AlchemyPotion(@NotNull String potionConfigName, @NotNull ItemStack potion
     }
 
     public boolean isSimilarPotion(@NotNull ItemStack otherPotion) {
+        return isSimilarPotion(otherPotion, otherPotion.getItemMeta());
+    }
+
+    public boolean isSimilarPotion(@NotNull ItemStack otherPotion, @Nullable ItemMeta otherMeta) {
         requireNonNull(otherPotion, "otherPotion cannot be null");
+
         if (otherPotion.getType() != potionItemStack.getType()) {
             return false;
         }
 
         // no potion meta, no match
-        if (!otherPotion.hasItemMeta()) {
+        if (otherMeta == null) {
```

Comment:
Don't understand why this change is necessary, isn't it equivalent to the prior version?

Reply:
It is equivalent, but iirc I added that check because IDEA complains that you are using a maybe null value in `hasDifferingCustomEffects`... or maybe it was my Kotlin brain saying that I should add a null check.

Reply:
![image](https://github.com/user-attachments/assets/baa25268-b2c9-41bd-881f-e324dd6c2beb)